### PR TITLE
fix(kuma-cp): add latest version to compat matrix

### PR DESCRIPTION
**Pull request by parkanzky**
Signed-off-by: Paul Parkanzky <paul.parkanzky@konghq.com>

### Summary

GUI still uses the compatibility matrix to find the latest Kuma DP version. Need to add version so that 1.6.0 does not complain about DP.

### Full changelog

* Add latest DP version to compatibility matrix.
 
### Issues resolved

Fix https://github.com/kumahq/kuma-gui/issues/321

### Documentation

- [ ] Link to the website [documentation PR](https://github.com/kumahq/kuma-website/pull/XXX)

### Testing

- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual testing on Universal
- [x] Manual testing on Kubernetes

### Backwards compatibility

- [ ] Update [`UPGRADE.md`](/UPGRADE.md) with any steps users will need to take when upgrading.
- [ ] Add `backport-to-stable` label if the code follows our [backporting policy](/CONTRIBUTING.md#backporting)

### CODE REVIEW

**Comment by lahabana** [comment]
I'm a little confused I thought this was not in use anymore why is this still in the code base?

**Comment by parkanzky** [comment]
> I'm a little confused I thought this was not in use anymore why is this still in the code base?

@lahabana  I also thought it was not used anymore. However, the GUI code still pulls it down and checks the version of the DP against the list, which caused this bug when the list was not updated.

**Review by michaelbeaumont — APPROVED** [review]
I don't know if anyone was really quite aware we were serving the matrix from the API server...

File: pkg/version/compatibility.go [hunk 1]
```diff
@@ -52,10 +52,13 @@ var CompatibilityMatrix = Compatibility{
 		"~1.5.0": {
 			Envoy: "~1.21.1",
 		},
+		"~1.6.0": {
+			Envoy: "~1.21.1",
+		},
 		// This includes all dev versions branched from the first release
 		// candidate (i.e. both master and release-1.4)
 		// and all 1.4 releases and RCs. See Masterminds/semver#21
```
[the remark this answers is not in the document]
Comment (michaelbeaumont) [reply pkg/version/compatibility.go:60]: Feel free to ignore this if the plan in the short/medium term is to remove this code/the endpoint from the API server

**Comment by lahabana** [comment]
Is the right fix to have this shipped only in release-1.6 and remove it in master then?

**Comment by parkanzky** [comment]
> Is the right fix to have this shipped only in release-1.6 and remove it in master then?

I think we should remove it as part of the task that replaces it.

**Comment by parkanzky** [comment]
@mergify backport release-1.6

**Comment by parkanzky** [comment]
@mergify backport tests/test-mergify-dco

**Pull request merged**